enterprise/cluster/src/main/java/org/neo4j/cluster/com/NetworkSender.java

```diff
@@ -424,6 +426,8 @@ public ChannelPipeline getPipeline() throws Exception
     private class NetworkMessageSender
             extends SimpleChannelHandler
     {
+        private Throwable lastException;
```

Comment:
I looked at CappedOperation, and it is not suitable in this situation. It does not have a reset method. I talked to @tinwelint and we decided to revisit CappedOperation in the future, to make it usable here.

What makes you thing exceptionCaught and writeComplete are called by different threads?